frontend/src/lib/StaticGraphParser.ts

```diff
@@ -68,7 +68,7 @@ export function _populateInfoFromTemplate(info: SelectedNodeInfo, template?: Tem
 
   if (template.inputs) {
     info.inputs =
-      (template.inputs.parameters || []).map(p => [p.name, p.value || '']);
+      (template.inputs.parameters || []).map(p => [p.name, p.value || ''] as [string, string]);
```

Comment:
nit: Can you give [string, string] a proper `type/interface` name and avoid casting here? Avoiding type casting helps with long term maintenance.

Reply:
One way to avoid casting here is
```
....map((p): [string, string] => [p.name, p.value || '']);
```
[example](http://www.typescriptlang.org/play/?ts=3.1.6#code/MYewdgzgLgBAhgJwQLhgQSXAngHgNrQICWYA5gDQyEmkC6AfDALwBQM7MeAjLQHQC2cAA4AKEQEpkBKMTKVqZWs0Z4A5HABGwVZVUATAKYAzVbXEsgA)

What would be even better, we should probably upgrade typescript version to at least 3.5, so it can handle it by default. [example](http://www.typescriptlang.org/play/?ts=3.5.1#code/MYewdgzgLgBAhgJwQLhgQSXAngHgNrQICWYA5gDQyEmkC6AfDALwBQM7MeAjLQHQC2cAA4AKEQEpmjPAHI4AI2AzKMgCYBTAGYza4lkA)

Reply:
updated typescript to 3.6.

Reply:
fixed issues introduced by the stricter typing by typescript >=3.5.

updated node to LTS 10 - because I notice a few peculiarity:

- the grpc packages are not found in `package-lock.json` even thou it specified in `package.json` (I think is subsequently installed in the post-install step, but I am not sure if it is used in any other places) - so whenever I do a `npm i` they get included into the lock file. And this result in some issues for node 9.4, as some dependent packages seems to be missing (does not support node 9)? So I have to upgrade the docker images to node 10.

- Some of packages are not following the semantic versioning faithfully - I have to freeze a couple of packages, as the latest ones breaks the app (when `package-lock.json` is updated when I upgraded `typescript`.

Reply:
> the grpc packages are not found in package-lock.json even thou it specified in package.json (I think is subsequently installed in the post-install step, but I am not sure if it is used in any other places) - so whenever I do a npm i they get included into the lock file. And this result in some issues for node 9.4, as some dependent packages seems to be missing (does not support node 9)? So I have to upgrade the docker images to node 10.

Thanks for the catch. I think the last committer forgot to update package-lock.json last time, let's just commit them so they are in sync. Besides upgrading to node 10 LTS is definitely a good choice.

> Some of packages are not following the semantic versioning faithfully - I have to freeze a couple of packages, as the latest ones breaks the app (when package-lock.json is updated when I upgraded typescript.

Thanks for finding them, the solution sounds good to me.